sdccc/src/main/java/com/draeger/medical/sdccc/messages/MessageStorage.java

```diff
@@ -310,7 +315,28 @@ protected MessageContent convertMessageToMessageContent(final Message message) {
                 mdibVersionGroups,
                 actions,
                 message.getID(),
-                isSOAP);
+                isSOAP,
+                getSender(message));
+    }
+
+    private String getSender(final Message message) {
+        final CommunicationContext communicationContext = message.getCommunicationContext();
+        if (communicationContext == null) {
+            return UNKNOWN_SENDER;
+        }
+        final TransportInfo transportInfo = communicationContext.getTransportInfo();
+        if (transportInfo == null) {
+            return UNKNOWN_SENDER;
+        }
+        if (message.getDirection() == CommunicationLog.Direction.INBOUND) {
+            final Optional<String> remoteAddress = transportInfo.getRemoteAddress();
+            if (remoteAddress.isEmpty()) {
+                return UNKNOWN_SENDER;
+            }
+            return remoteAddress.orElseThrow();
+        } else {
```

Comment:
Please explicitly check for the outbound direction and treat other cases as errors.

Reply:
Done.